crates/sargon-uniffi/src/signing/signatures_per_fractor_source.rs

```diff
@@ -0,0 +1,34 @@
+use crate::prelude::*;
+use paste::paste;
+
+macro_rules! decl_signatures_per_factor_source {
+    ($signable_id:ty) => {
+        paste! {
+            use sargon::[< $signable_id >] as [< Internal $signable_id >];
```

Comment:
minor improvment is to create an inner clause of this macro, and let the outer call the inner, with the `[< HDSignatureOf $signable_id >]` as an argument, so that you dont need to specify `[< HDSignatureOf $signable_id >]` *and** `[< SignaturesPerFactorSourceOf $signable_id >]` twice like this. It might read out better. Like so:

```rust
macro_rules! decl_signatures_per_factor_source {
    (
        struct: $struct_name:ident, 
        of: $item:ident
    ) => {
        #[derive(Clone, PartialEq, Eq, uniffi::Record)]
        pub struct $struct_name {
            pub factor_source_id: FactorSourceIDFromHash,
            pub hd_signatures: Vec<$item>
        }

        impl $struct_name {

            pub fn new(
                factor_source_id: FactorSourceIDFromHash,
                hd_signatures: Vec<$item>
            ) -> Self {
                Self {
                    factor_source_id,
                    hd_signatures
                }
            }

        }
    };
    ($signable_id:ty) => {
        paste! {
            use sargon::[< $signable_id >] as [< Internal $signable_id >];
            decl_signatures_per_factor_source!(
                struct: [< SignaturesPerFactorSourceOf $signable_id >],
                of: [< HDSignatureOf $signable_id >]
            );
        }
    };
}

decl_signatures_per_factor_source!(TransactionIntentHash);
decl_signatures_per_factor_source!(SubintentHash);
```

You can use this technique for your other macros too perhaps - if you like it. I prefer it because I think `paste!` brings some "noise".

Reply:
Will keep that in mind.